apps/blockchain/lib/blockchain/block.ex

```diff
@@ -710,40 +711,45 @@ defmodule Blockchain.Block do
   in `:ets` or `:rocksdb`. See `MerklePatriciaTree.DB`.
   """
   @spec add_transactions(t, [Transaction.t()], DB.db(), Chain.t()) :: t
-  def add_transactions(block, transactions, db, chain) do
-    block
-    |> process_hardfork_specifics(chain, db)
-    |> do_add_transactions(transactions, db, chain)
-    |> calculate_logs_bloom()
+  def add_transactions(block, transactions, trie, chain) do
+    {updated_block, updated_trie} = process_hardfork_specifics(block, chain, trie)
+
+    {updated_block, updated_trie} =
+      do_add_transactions(updated_block, transactions, updated_trie, chain)
```

Comment:
If we updated the arguments that `do_add_transactions` and `calculate_logs_bloom` expects I think this could become:

```ex
def add_transactions(block, transactions, trie, chain) do
  process_hardfork_specifics(block, chain, trie)
  |> do_add_transactions(transactions, chain)
  |> calculate_logs_bloom
end

def do_add_transactions({updated_block, updated_trie}, transactions, chain) do
  # ...
end

def calculate_logs_bloom({updated_block, updated_trie}) do
  # ...
  {updated_block, updated_trie}
end
```

Reply:
I will update it in the next PR